File: OPENAPI-REST-API/openapi-client/scalaz/src/main/scala/org/openapitools/client/api/TicketPostDetailsInner.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import TicketPostDetailsInner._

case class TicketPostDetailsInner (
  post_id: Option[Integer],
date: Option[String],
contents: Option[String],
creator: Option[Creator],
creator_email: Option[String],
creator_name: Option[String],
hasattachments: Option[Integer],
attachment_download: Option[String])

object TicketPostDetailsInner {
  import DateTimeCodecs._
  sealed trait Creator
  case object User extends Creator
  case object Staff extends Creator

  object Creator {
    def toCreator(s: String): Option[Creator] = s match {
      case "User" => Some(User)
      case "Staff" => Some(Staff)
      case _ => None
    }

    def fromCreator(x: Creator): String = x match {
      case User => "User"
      case Staff => "Staff"
    }
  }

  implicit val CreatorEnumEncoder: EncodeJson[Creator] =
    EncodeJson[Creator](is => StringEncodeJson(Creator.fromCreator(is)))

  implicit val CreatorEnumDecoder: DecodeJson[Creator] =
    DecodeJson.optionDecoder[Creator](n => n.string.flatMap(jStr => Creator.toCreator(jStr)), "Creator failed to de-serialize")

  implicit val TicketPostDetailsInnerCodecJson: CodecJson[TicketPostDetailsInner] = CodecJson.derive[TicketPostDetailsInner]
  implicit val TicketPostDetailsInnerDecoder: EntityDecoder[TicketPostDetailsInner] = jsonOf[TicketPostDetailsInner]
  implicit val TicketPostDetailsInnerEncoder: EntityEncoder[TicketPostDetailsInner] = jsonEncoderOf[TicketPostDetailsInner]
}

File: OPENAPI-REST-API/openapi-client/zapier/models/TicketPostDetails_inner.js
const utils = require('../utils/utils');

module.exports = {
    fields: (prefix = '', isInput = true, isArrayChild = false) => {
        const {keyPrefix, labelPrefix} = utils.buildKeyAndLabel(prefix, isInput, isArrayChild)
        return [
            {
                key: `${keyPrefix}post_id`,
                label: `[${labelPrefix}post_id]`,
                type: 'integer',
            },
            {
                key: `${keyPrefix}date`,
                label: `[${labelPrefix}date]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}contents`,
                label: `[${labelPrefix}contents]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}creator`,
                label: `[${labelPrefix}creator]`,
                type: 'string',
                choices: [
                    'User',
                    'Staff',
                ],
            },
            {
                key: `${keyPrefix}creator_email`,
                label: `[${labelPrefix}creator_email]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}creator_name`,
                label: `[${labelPrefix}creator_name]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}hasattachments`,
                label: `[${labelPrefix}hasattachments]`,
                type: 'integer',
            },
            {
                key: `${keyPrefix}attachment_download`,
                label: `[${labelPrefix}attachment_download]`,
                type: 'string',
            },
        ]
    },
    mapping: (bundle, prefix = '') => {
        const {keyPrefix} = utils.buildKeyAndLabel(prefix)
        return {
            'post_id': bundle.inputData?.[`${keyPrefix}post_id`],
            'date': bundle.inputData?.[`${keyPrefix}date`],
            'contents': bundle.inputData?.[`${keyPrefix}contents`],
            'creator': bundle.inputData?.[`${keyPrefix}creator`],
            'creator_email': bundle.inputData?.[`${keyPrefix}creator_email`],
            'creator_name': bundle.inputData?.[`${keyPrefix}creator_name`],
            'hasattachments': bundle.inputData?.[`${keyPrefix}hasattachments`],
            'attachment_download': bundle.inputData?.[`${keyPrefix}attachment_download`],
        }
    },
}
